### DISCUSSION TITLE
Support scope as an array

### DISCUSSION THREAD

**Discussion by balazsorban44** [panva/openid-client#385] [Ideas]
Thank you for this awesome library!

Currently, the `scope` in `AuthorizatinoParameters` can only be of type `string`.
https://github.com/panva/node-openid-client/blob/772306cc82b72ace9361f4a945249a5c35a8f921/types/index.d.ts#L138

This is merely a suggestion, but since the `claims` property has also special handling, I was wondering if we could make an exception for `scope` as well since it is an often-used parameter.

Related: https://github.com/nextauthjs/next-auth/pull/2511#issuecomment-897503785

**I am happy to open a PR implementing this if the author approves of the idea!**

Relevant code is probably here:

https://github.com/panva/node-openid-client/blob/772306cc82b72ace9361f4a945249a5c35a8f921/lib/client.js#L84-L86

I have seen scopes being sent with separators like ` `, `+`, `,`. `node-oauth` uses [`querystring.stringify()`](https://nodejs.org/api/querystring.html#querystring_querystring_stringify_obj_sep_eq_options):
https://github.com/ciaranj/node-oauth/blob/a7f8a1e21c362eb4ed2039431fb9ac2ae749f26a/lib/oauth2.js#L174
, which produces something like this: `?scope=openid&scope=email`

`new URLSearchParams({scope: ["openid", "email"]})` produces `,`.

Is there any consequence to choose any concatenating over the other? Is this something that can be interpreted differently from Provider to Provider and so that is the reason scope as an array isn't supported?

**Comment by panva** [comment]
Feel free to propose a PR, the only correct way is to `.join()` the array with a space `' '` as the separator. Everything else is just plain wrong ;)

- `scope=openid&scope=email` - wrong because it submits duplicate parameters and rejected by a conform AS
- `scope=openid,email` - wrong because it is parsed as a single scope value by a conform AS

**Reply by balazsorban44** [reply]
Thank you! Will have a look then!
